src/box/relay.cc

```diff
@@ -272,6 +273,7 @@ relay_new(struct replica *replica)
 	assert(relay != NULL);
 
 	memset(relay, 0, sizeof(struct relay));
+	iostream_clear(&relay->io);
```

Comment:
I need a way to create an unusable object that can only be used for properly initializing it later (with `iostream_create`) and checking if it was initialized (with `iostream_is_initialized`). If it was a pointer, I would set it to NULL, but it isn't. Options I see:

1. `iostream_is_initialized` checks vtab instead of fd. Then we could use it after memset. However, users may erroneously access fd. If it's 0, it's a valid fd so a bug in the code could result in undefined behavior. We could wrap fd access in a helper function (`iostream_fd`), but I'd rather not - accessing fd directly is more convenient IMO.
2. Rename `iostream_clear` to `iostream_create` and add a method to finalize initialization (`iostream_set_fd`?). Not all users need to create an unusable object, but this would force those users call two methods instead of one.
3. Allow to pass -1 to `iostream_create`:
   ```C
   iostream_create(io, -1);
   /* ... */
   iostream_create(io, fd);
   ```
   That is, there would be two constructor calls without a destruction, which is confusing.

Think of `iostream_clear` as `memset(p, 0, size)`. It's optional, but sometimes it can be useful.